plugins/metrics/pkg/agent/runner.go

```diff
@@ -195,18 +191,25 @@ func (tr *taskRunner) doPush(ctx context.Context, writeRequest *prompb.WriteRequ
 			default:
 			}
 
-			if !isRecoverable(err) {
+			switch {
+			case strings.Contains(err.Error(), "context cancelled"),
+				strings.Contains(err.Error(), "ingestion rate limit"):
+
+				tr.logger.With(
+					zap.Error(err),
+				).Warn("failed to push to remote write, retrying...")
+			default:
 				return fmt.Errorf("failed to push to remote write: %w", err)
 			}
-
-			tr.logger.With(
-				zap.Error(err),
-			).Error("failed to push to remote write, retrying...")
 		}
 	}
 }
 
 func (tr *taskRunner) OnTaskRunning(ctx context.Context, activeTask task.ActiveTask) error {
+	limit := WriteLimit{
+		GrpcMaxBytes:             4194304,
+		CortexIngestionRateLimit: 2500,
```

Comment:
The default rate limit is 25000 samples/s, not 2500. That said, the agent should probably not need to worry about ingestion rate limits, and just handle the 429 status appropriately.

Reply:
Ooops, typo. I think in the case where a write request has more samples than can fit in the ingestion rate limit we do need to worry about the limits. If the request itself is too large to fit within that window we'll just keep entering the backoff.

Reply:
We should make the limit values configurable though, and part of the import spec.

Reply:
That's something I am working on right now, you'll be able to configure limits globally and also per-tenant as part of the cortexops api.